mantis-common/src/main/java/io/mantisrx/common/WorkerPorts.java

```diff
@@ -43,39 +52,29 @@ public WorkerPorts(final List<Integer> assignedPorts) {
         this.debugPort = assignedPorts.get(1);
         this.consolePort = assignedPorts.get(2);
         this.customPort = assignedPorts.get(3);
-        this.ports = new ArrayList<>(3);
-        for (int idx = 4; idx < assignedPorts.size(); idx++) {
-            ports.add(assignedPorts.get(idx));
-
-            sinkPort = assignedPorts.get(idx);
-            break;
-        }
+        this.sinkPort = assignedPorts.get(4);
+        this.ports = new ArrayList<>(1);
+        ports.add(assignedPorts.get(4));
 
         if (!isValid()) {
             throw new IllegalStateException("worker validation failed on port allocation");
         }
     }
 
     public WorkerPorts(int metricsPort, int debugPort, int consolePort, int customPort, int sinkPort) {
-        this.ports = new ArrayList<>(5);
```

Comment:
could you pl refactor this to call the `WorkerPorts(List<Integer>);` constructor?
I'd also make `ports` an immutable because we return the value as-is in `getPorts`.

Reply:
+1